instant-segment/examples/merge.rs

```diff
@@ -0,0 +1,246 @@
+//! Merge downloaded data to synthesize test data files
+
+use std::cmp::Reverse;
+use std::fs::File;
+use std::io::Write;
+use std::io::{BufRead, BufReader, BufWriter};
+use std::str::FromStr;
+
+use ahash::{AHashMap as HashMap, AHashSet as HashSet};
+use rayon::iter::{IntoParallelIterator, ParallelIterator};
+use smartstring::alias::String as SmartString;
+
+fn main() {
+    let word_list = read_word_list();
+    process_unigrams(&word_list);
+    process_bigrams(&word_list);
+}
+
+/// Read bigrams from the input file parts, filter them, and write to file
+fn process_bigrams(word_list: &HashSet<SmartString>) {
+    let bigrams = (0..BIGRAM_PARTS)
+        .into_par_iter()
+        .map(|part| {
+            let fname = format!("data/cache/eng-2-{:05}-{:05}.txt", part, BIGRAM_PARTS);
+            let f = File::open(&fname).unwrap();
+            let mut reader = BufReader::with_capacity(4 * 1024 * 1024, f);
+
+            let mut ln = String::new();
+            let mut bigrams = HashMap::new();
+            loop {
+                ln.clear();
+                match reader.read_line(&mut ln) {
```

Comment:
Could also give an example of what a line looks like here.

Reply:
Done.